src/node/rpc/member_frontend.h

```diff
@@ -192,6 +227,96 @@ namespace ccf
       }
     }
 
+    bool deploy_js_app(kv::Tx& tx, std::string name, const JsBundle& bundle)
+    {
+      std::regex name_pattern("[a-zA-Z0-9_-]+");
+      if (!std::regex_match(name, name_pattern))
+      {
+        LOG_FAIL_FMT("app name is invalid");
+        return false;
+      }
+
+      std::string module_prefix = fmt::format("/{}/", name);
+      std::string url_prefix = name;
+
+      remove_modules(tx, module_prefix);
+      set_modules(tx, module_prefix, bundle.modules);
+
+      remove_endpoints(tx, url_prefix);
+
+      auto tx_scripts = tx.get_view(network.app_scripts);
+      for (auto& [url, endpoint] : bundle.metadata.endpoints)
+      {
+        for (auto& [method, info] : endpoint)
+        {
+          const std::string& js_module = info.js_module;
+          if (std::none_of(
+                bundle.modules.cbegin(),
+                bundle.modules.cend(),
+                [&js_module](const SetModule& item) {
+                  return item.name == js_module;
+                }))
+          {
+            LOG_FAIL_FMT(
+              "{} {}: module '{}' not found in bundle",
+              method,
+              url,
+              info.js_module);
+            return false;
+          }
+
+          // CCF currently requires each endpoint to have an inline JS module.
+          std::string method_uppercase = method;
+          nonstd::to_upper(method_uppercase);
+          std::string key =
+            fmt::format("{} {}{}", method_uppercase, url_prefix, url);
+          std::string script = "import {" + info.js_function + " as f}" +
```

Comment:
Can you build this with `fmt::format`?

Reply:
If you tell me how to escape curly braces, then yes :) Keep in mind that this is temporary code that will hopefully be killed soon, and it's not in the critical path.

Reply:
I believe its `{{` to produce a literal `{`.